src/graph/transform/to_block.cc

```diff
@@ -144,6 +145,174 @@ std::tuple<HeteroGraphPtr, std::vector<IdArray>> ToBlockCPU(
 
 }  // namespace
 
+template <typename IdType>
+std::tuple<HeteroGraphPtr, std::vector<IdArray>> ToBlockProcess(
```

Comment:
this will be shared with existing `ToBlockCPU()` in another PR?

Reply:
Yes,  in #5297